test/serializer/abstract/UpdateExpression3.js

```diff
@@ -0,0 +1,4 @@
+foo = {x: 42};
+let obj = global.__abstract ? __abstract({x: __abstract('number', 'foo.x')}, "foo") : foo;
```

Comment:
Added some complex tests.

What's the best way to define an abstract object that preserves its identity in a test?